keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -0,0 +1,435 @@
+    // A pod will be in this map from the time when the API server processed the
+    // eviction request to the time when the pod is seen by PDB controller
+    // as having been marked for deletion (or after a timeout). The key in the map is the name of the pod
+    // and the value is the time when the API server processed the eviction request. If
+    // the deletion didn't occur and a pod is still there it will be removed from
+    // the list automatically by PodDisruptionBudget controller after some time.
+    // If everything goes smooth this map should be empty for the most of the time.
+    // Large number of entries in the map may indicate problems with pod deletions.
+    // +optional
+    DisruptedPods map[string]metav1.Time
+
+    // Number of pod disruptions that are currently allowed.
+    DisruptionsAllowed int32
+
+    // current number of healthy pods
+    CurrentHealthy int32
+
+    // minimum desired number of healthy pods
+    DesiredHealthy int32
+
+    // total number of pods counted by this disruption budget
+    ExpectedPods int32
+
+    // Represents the latest available observations of a pdb's current state.
+    Conditions []DeploymentCondition
+}
+```
+
+### Test Plan
+
+#### Existing Tests
+PodDisruptionBudget currently has tests in various components that use the feature:
+
+* Unit tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/disruption/disruption_test.go
+* Integration tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/test/integration/disruption/disruption_test.go
+* Kubectl
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate/versioned/pdb_test.go
+  tests generation of a PDB objects out of given parameters
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/create/create_pdb_test.go
+  tests creation of PDB objects from cmd parameters
+* Scheduler
+  - https://github.com/kubernetes/kubernetes/blob/4b59044b8d2a3502ea490ba2c958008a098511a3/test/integration/scheduler/preemption_test.go#L1053
+  tests effects of PDB on preemption (PDB is honored in a best effort way)
+* Eviction integration tests
+  - https://github.com/kubernetes/kubernetes/blob/master/test/integration/evictions/evictions_test.go test eviction logic and its interactions with PDB.
+* Autoscaler
+  - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/autoscaling/cluster_size_autoscaling.go ensure that Autoscaler respects PDB when draining nodes.
+* Integration tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/disruption.go
+
+All beta PDB endpoints are covered by e2e tests: https://apisnoop.cncf.io/1.20.0/beta/policy
+
+New tests will be added to test the planned changes. This includes:
+* Verify that an empty selector has the correct semantics
+* Verify that pods with incompatible controllers are handled correctly and that
```

Comment:
some test (e2e? integration?) that checks if PDB behaves as expected if it has pods both controlled by object with scale/ subresource and ones without that (with percentage MinAvailable)

?

Reply:
Added some more information here about the scenarios that needs to be covered by the tests.